app/core/setup/setupSecrets.js

```diff
@@ -0,0 +1,20 @@
+const config = require('config');
+const { get, set } = require('lodash');
+
+const setSecret = (secretPath, configPath) => {
+  // Only overwrite the value if the secretPath is defined
+  if (config.has(secretPath)) {
+    set(config, configPath, get(config, secretPath));
+  }
+};
+
+const setup = () => {
+  if (config.has('secrets.ccd')) {
+    setSecret('secrets.ccd.ccd-api-gateway-oauth2-client-secret', 'secrets.ccd.ccd-api-gateway-oauth2-client-secret');
```

Comment:
these mappings look wrong to me. if they are same on both sides, then we don't need this.

Reply:
they are the same on both sides, https://github.com/hmcts/ccd-api-gateway/blob/master/config/default.yaml#L31-L34

shouldn't be needed, except for app insights key, and given it's only one property i would update the app insights key and get rid of this file,

that would also mean you don't need to fix the test failure that occurred when you introduced the test for this